designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/BaseEndpointDefinitions.scala

```diff
@@ -86,9 +87,24 @@ object BaseEndpointDefinitions {
 
 }
 
-sealed trait SecurityError
+sealed trait NuError
+case class NuException(nuError: NuError) extends Exception
+
+sealed trait SecurityError extends NuError
 
 object SecurityError {
   case object AuthenticationError extends SecurityError
   case object AuthorizationError  extends SecurityError
 }
+
+sealed trait BusinessError extends NuError {
```

Comment:
This may not be a good approach. As I see this is a global Business error. And it's sealed. Currently, it's related to Scenario Activity API. Let's imagine in the future we will identify a new business error and add it here. In the Scenario Activity HTTP Service we will be forced to handle the new case. But the case won't be related to the Scenario Activity API.

That's why IMO, we should follow the approach from AppApiHttpService, when the business errors are defined locally.

Reply:
Done